### PR TITLE
Non exhaustive

### LINKED ISSUE #21: use #[non_exhaustive] for config
This would allow to add fields without breaking changes

suggested as option (1) in https://github.com/bitcoindevkit/bdk/pull/429#issuecomment-908973821

### PR DESCRIPTION
**Pull request by RCasatta**
fix #21 

on top of #22

### CODE REVIEW

**Comment by RCasatta** [comment]
It looks this cause trouble if downstream crates use

```
let conf = bitcoind::Conf {
            view_stdout: true,
            ..Default::default()
        };
```

```
error[E0639]: cannot create non-exhaustive struct using struct expression
```

**Comment by RCasatta** [comment]
Given that the syntax `....Default::default()` it's not supported with non-exhaustive, so user's code should do something like:

```
let mut conf = bitcoind::Conf::default();
conf.view_stdout = true;
```

Is this desirable for end-users? Or it's better to not have this (and `non_exhaustive`) at the cost of breaking downstream when new parameters are added?

I think `non_exhaustive` is still desirable (maybe coupled with a builder pattern)
What @LeoComandini @LLFourn @rajarshimaitra think?

File: src/lib.rs [hunk 1]
```diff
@@ -55,6 +55,7 @@ pub struct ConnectParams {
 }
 
 /// Enum to specify p2p settings
+#[derive(Debug, PartialEq)]
```
Comment (LeoComandini) [inline src/lib.rs:58]: Should we add `Eq` as well?

Comment (RCasatta) [reply src/lib.rs]: added

**Comment by LeoComandini** [comment]
utACK df37667072325c0210dff3787c0a2eebadab4abe, I think `non_exhaustive` is desirable

**Comment by Kixunil** [comment]
FYI from my experience for library *users* it's *usually* most convenient to use builder pattern which moves `self`. Although this is arguably annoying to implement. I think there's a proc macro for that.

I think it'd be perfect if either we could do `non_exhaustive` with `..Default::default()` or have some sort of default constructor and then struct initialization syntax just overwrites the fields.

**Pull request merged**